docker/developers/Dockerfile

```
@@ -0,0 +1,203 @@
+# Web Environment
+# ----------------
+
+# Ubuntu 20.04
+FROM ubuntu:focal AS cdo-web
+
+ARG USERNAME=cdodev
+
+ARG DEBIAN_FRONTEND=noninteractive
+ENV TZ=America/Los_Angeles
+
+# Additional required packages
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends ca-certificates \
+    && apt-get -y install --no-install-recommends \
+    autoconf gcc g++ bison build-essential libssl-dev libyaml-dev libreadline-dev \
+    zlib1g-dev libncurses-dev libffi-dev git git-lfs parallel debian-archive-keyring
+
+# Set the locale
+RUN apt-get -y install locales
+RUN sed -i '/en_US.UTF-8/s/^# //g' /etc/locale.gen && \
+    locale-gen
+ENV LANG en_US.UTF-8  
+ENV LANGUAGE en_US:en  
+ENV LC_ALL en_US.UTF-8 
+
+# Set the non-root user for the container and switch
+ARG UID
+ARG GID
+RUN apt-get -y install sudo
+RUN groupadd -g ${GID} ${USERNAME} \
+    && useradd -r -u ${UID} -g ${USERNAME} --shell /bin/bash --create-home ${USERNAME} \
+    && echo "${USERNAME} ALL=NOPASSWD: ALL" >> /etc/sudoers \
+    && chown -R ${USERNAME} /usr/local
+USER ${USERNAME}
+
+# Install ImageMagick
+RUN sudo -E apt-get -y install --no-install-recommends libmagickwand-dev imagemagick
+
+# MySQL Dependencies
+RUN sudo -E apt-get -y install --no-install-recommends cmake wget libaio-dev
+
+# Install MySQL Native Client
+RUN sudo -E apt-get -y install --no-install-recommends mysql-client-8.0 libmysqlclient-dev
+
+# Install rbenv
+RUN sudo -E apt-get -y install rbenv \
+    && sudo mkdir -p "/opt/plugins" \
+    && mkdir -p /home/${USERNAME}/.rbenv/plugins \
+    && sudo chown -R ${USERNAME} /opt/plugins \
+    && git clone https://github.com/rbenv/ruby-build.git /opt/plugins/ruby-build \
+    && ln -s /opt/plugins/ruby-build /home/${USERNAME}/.rbenv/plugins/ruby-build
+
+# Install Ruby. Also replace the system ruby (required for RubyMine debugging).
+ARG RUBY_VERSION=3.0.5
+RUN rbenv install ${RUBY_VERSION} \
+    && echo -n '\n# rbenv init\neval "$(rbenv init -)"\n' >> ~/.bashrc \
+    && rbenv global ${RUBY_VERSION} \
+    && sudo rm /usr/bin/ruby \
+    && sudo ln -s /home/${USERNAME}/.rbenv/versions/${RUBY_VERSION} /usr/bin/ruby
+
+# Install nvm
+ARG NODE_VERSION=18.16.0
```

I don't believe we are pinned to a specific minor version of node, so we can use `18` for `NODE_VERSION` ( I didn't see the equivalent for NVM though)

```suggestion
ARG NODE_VERSION=18
```

---

`SETUP.md` specifies a specific version, so the Dockerfile does, too.